proposals/wg-image-compatibility.md

```diff
@@ -0,0 +1,111 @@
+# OCI Working Group Proposal: Image Compatibility
+
+Proposal created from [OCI WG template](https://github.com/opencontainers/tob/blob/master/WG-TEMPLATE.md).
+
+## Image Compatibility OCI Working Group - Governance Charter
+
+This document describes the basic governance principles for the Image Compatibility Working Group (the “WG”).
+
+The WG operates as an OCI Working Group under the [Open Container Initiative (OCI) Charter](https://github.com/opencontainers/tob/blob/master/CHARTER.md), which describes the responsibilities of the OCI Technical Oversight Board (the "TOB”).
+The WG is established by the TOB as an OCI Working Group pursuant to the OCI Charter.
+Accordingly, the WG will operate in accordance with the OCI Charter and OCI's other policies and procedures, supplemented by the details below.
+
+## Purpose
+
+In mission-critical industries, applications often require special features provided by host operating systems.
+These are usually described in the release notes or installation guides.
```

Comment:
NFD describes the node side. We want to provide a possibility to describe container requirements, the container side.

Reply:
NFD also mentioned here https://docs.google.com/document/d/1lzwh8DGMu5vXXHwJmnewYIMffkcOEvH8owX4UYjRcw0/edit?disco=AAAA5ncnzqM 

The image compatibility is to describe what special needs to the host OS by the image author from container application perspective, but not what the node features exist. For example, for free 5GC UPF, https://free5gc.org/guide/3-install-free5gc/, kernel module gtp5g and kernel versions are required to run UPF normally.

Yeah, there may be some overlap with NFD reporting in concepts(or objects). For k8s use case, NFD and image compatibility could be combined to make appropriate scheduling decisions if needed.

But the image compatibility is not only for Kubernetes use cases. It can be discussed in the working group, whether some fields in NFD are generic enough and can be re-used in image compatibility.